src/Void.Core/CommandMode.fs

```diff
@@ -8,13 +8,14 @@ module CommandMode =
         | EntryCancelled
         | CharacterBackspaced
         | TextAppended of string
-        | CommandCompleted
+        | TextReplaced of string
+        | CommandCompleted of string
         interface EventMessage
 
     let private handleEnter interpret buffer =
         match interpret { Language = "VoidScript"; Fragment = buffer} with
         | InterpretScriptFragmentResponse.Completed ->
-            ("", Event.CommandCompleted :> Message)
+            ("", Event.CommandCompleted buffer :> Message)
         | InterpretScriptFragmentResponse.ParseFailed error ->
             ("", CoreEvent.ErrorOccurred error :> Message)

```

Comment:
I've gotta figure out how to get these to CommandHistory and strip out the attempted command (at least for unknown command errors)